### PR TITLE
[SceneCreator] Add New methods

### PR DESCRIPTION
**Pull request by epernod**
Small (and first) Pullrequest to add several methods in plugin SceneCreator:

- to add basic 3D object: Cube, Cylinder, Plane. In rigid or deformable.
- also add an executable sceneCreatorExamples to run simple examples using those methods.
- Minor changes in TriangularFEM and CylindricalGrid to use those components.

 This PR: 
  - [x] succeed on each compilation setup on the CI
  - [x] does not generates new warnings or test failure
  - [x] does not seems to break existing scenes.
  - [x] does not seems to break API compatibility.  
  - [x] clean stuff
  - [x] is now 1 week old and no one send a 'no go' comment.   
  - [x] have tests on the introduced features.

### CODE REVIEW

File: applications/plugins/SceneCreator/SceneCreator.cpp [hunk 1]
```diff
@@ -329,12 +345,26 @@ Node::SPtr createVisualNodeVec3(Node::SPtr  parent, MechanicalObject3::SPtr  dof
     visual->setRotation(rotation[0],rotation[1],rotation[2]);
     VisualNode->addObject(visual);
 
-    BarycentricMapping3_to_Ext3::SPtr mapping = New<BarycentricMapping3_to_Ext3>();
-    mapping->setModels(dof.get(), visual.get());
-    mapping->setName("Mapping Visual");
-    mapping->setPathInputObject(refDof);
-    mapping->setPathOutputObject(refVisual);
-    VisualNode->addObject(mapping);
+    if (mappingT == MT_Barycentric) // TODO check if possible to create a baseMapping::SPtr before the switch
```
Comment (damienmarchal) [inline applications/plugins/SceneCreator/SceneCreator.cpp]: Can you be more specific with this TODO ? Do you intend to really make it ?

File: applications/plugins/SceneCreator/SceneCreator.cpp [hunk 2]
```diff
@@ -329,12 +345,26 @@ Node::SPtr createVisualNodeVec3(Node::SPtr  parent, MechanicalObject3::SPtr  dof
...
+    {
+        IdentityMapping3_to_Ext3::SPtr mapping = sofa::core::objectmodel::New<IdentityMapping3_to_Ext3>();
+        mapping->setModels(dof.get(), visual.get());
+        mapping->setName("Mapping Visual");
+        mapping->setPathInputObject(refDof);
+        mapping->setPathOutputObject(refVisual);
+        VisualNode->addObject(mapping);
+    }
+    else
+        std::cout << "Error: mapping visual not possible";
```
Comment (damienmarchal) [inline applications/plugins/SceneCreator/SceneCreator.cpp]: Usage of std::cout is allowed only for debugging not for within a PR.
Please use msg_error(this) << "Mapping visual not possible"
Bonus point if you explain why it is not possible and how the user is supposed to fix his scene to remove the message.

Comment (epernod) [reply applications/plugins/SceneCreator/SceneCreator.cpp]: answer for the TODO: "I don't know if it is possible as it is a BaseMapping in sofa core. 
I was going to ask that in the next meeting."

ok  for the change.
Edit: msg_error(this) is not possible in this case as it isn't a method inside a class.

File: modules/SofaMiscFem/TriangularFEMForceField.h [hunk 3]
```diff
@@ -255,7 +257,16 @@ class TriangularFEMForceField : public core::behavior::ForceField<DataTypes>
     void setDamping(Real val) { f_damping.setValue(val); }
     int  getMethod() { return method; }
     void setMethod(int val) { method = val; }
-
+    void setMethod(std::string methodName) {
```
Comment (damienmarchal) [inline modules/SofaMiscFem/TriangularFEMForceField.h:260]: As the setMethod can fails, shouldn't it be more adequate to returns a boolean indicating it has succeeded or not ?

**Comment by damienmarchal** [comment]
Hi Erik,

Nice commit.

Some (cosmetic) remarks:
 - you updated some attributes/data field names but the name you use are not following the sofa guidelines. Data should starts with d_ not m_  (https://github.com/sofa-framework/sofa/blob/master/GUIDELINES.md). 
  - there are std::cout which should be replaced by the msg_error api 
 
+1 for the examples.
Would be nice to also have a test file validating the behavior.

File: modules/SofaMiscFem/TriangularFEMForceField.h (hunk 3 above)
Comment (epernod) [reply modules/SofaMiscFem/TriangularFEMForceField.h:260]: I agree, and there is more to do on those components, see #134

**Comment by damienmarchal** [comment]
Thanks for the changes, 

@epernod, do you have time to add tests to validate the new features you introduced ? 
(I know that you made examples but examples are not as informative as tests to detect regression)

**Comment by epernod** [comment]
@damienmarchal ok for the test. To have an example, could you point me out an example of test done in a plugin. Thanks.

**Comment by damienmarchal** [comment]
@epernod 
When making tests I generally try to tests either all the code-path or if this is too big, to test at least each added function with several values (eg: one with valid values and one with wrong one). 

For examples of tests on a Class you can have at the test added in: 
https://github.com/sofa-framework/sofa/pull/119/files

For examples of tests on a component you can have look at the last Eulalie's PR (
https://github.com/sofa-framework/sofa/pull/108)
and specifically at: 
https://github.com/SofaDefrost/sofa/blob/0df8da53458d64ed22b1cd2807555838e23ef0cf/SofaKernel/modules/SofaEngine/SofaEngine_test/BoxROI_test.cpp
you can also tests as in PR https://github.com/sofa-framework/sofa/pull/105 (interesting examples as they show how use ExpectMessage/MessageAsTestFailure to detect that a message has not been sended by the component)
and more specifically in: 
https://github.com/SofaDefrost/sofa/blob/65a113512104f0237a4c4bf5e6f05b2c1d6a7013/SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/DefaultPipeline_test.cpp




PS: I'm not saying this a super perfect tests but may gave you and idea on the direction to go.

**Comment by damienmarchal** [comment]
@epernod Did you managed to make some tests ? 

EDIT: up :)

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by bcarrez** [comment]
[ci-build]

**Comment by epernod** [comment]
back from a little "into the wild" break. Will finish that in the next days.

Edit: Test added. Not perfect but more will come as I add new methods in sceneCreator.

**Comment by epernod** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by epernod** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Pull request merged**